tests/utilTest.cpp

```diff
@@ -0,0 +1,421 @@
+
+    std::vector<bool> unevensize(4);
+    unsigned char abyte[1] = { 0x55 };
+    libzerocash::convertBytesToVector(abyte, unevensize);
+
+    v2 = { 0, 1, 0, 1 };
+    BOOST_CHECK(unevensize == v2);
+}
+
+BOOST_AUTO_TEST_CASE( testConvertVectorToBytes) {
+    unsigned char bytes[5] = {0x00, 0x01, 0x03, 0x12, 0xFF};
+    std::vector<bool> v = {
+        // 0x00
+        0, 0, 0, 0, 0, 0, 0, 0,
+        // 0x01
+        0, 0, 0, 0, 0, 0, 0, 1,
+        // 0x03
+        0, 0, 0, 0, 0, 0, 1, 1,
+        // 0x12
+        0, 0, 0, 1, 0, 0, 1, 0,
+        // 0xFF
+        1, 1, 1, 1, 1, 1, 1, 1
+    };
+    unsigned char output[5];
+    libzerocash::convertVectorToBytes(v, output);
+    BOOST_CHECK( memcmp(bytes, output, sizeof(bytes)) == 0 );
+
+    unsigned char onebyte[1];
+    std::vector<bool> unevensize { 1, 1, 1, 1, 1, 1, 1 };
+    libzerocash::convertVectorToBytes(unevensize, onebyte);
+    BOOST_CHECK(onebyte[0] == 127);
+}
+
+BOOST_AUTO_TEST_CASE( testConvertBytesToBytesVector ) {
+    unsigned char bytes[16];
+    for (int i = 0; i < 16; i++) {
+        bytes[i] = i;
+    }
+    std::vector<unsigned char> v(16);
+    libzerocash::convertBytesToBytesVector(bytes, v);
+    for (int i = 0; i < 16; i++) {
+        BOOST_CHECK(v.at(i) == bytes[i]);
+    }
+}
+
+BOOST_AUTO_TEST_CASE( testConvertBytesVectorToBytes ) {
+    std::vector<unsigned char>v(16);
+    for (int i = 0; i < 16; i++) {
+        v[i] = i;
+    }
+    unsigned char bytes[16];
+    memset(bytes, 0, 16);
+    libzerocash::convertBytesVectorToBytes(v, bytes);
+    for (int i = 0; i < 16; i++) {
+        BOOST_CHECK(bytes[i] == v.at(i));
+    }
+}
+
+BOOST_AUTO_TEST_CASE( testConvertBytesVectorToVector ) {
+    std::vector<unsigned char> bytes = {0x00, 0x01, 0x03, 0x12, 0xFF};
+    std::vector<bool> expected_bits = {
+        // 0x00
+        0, 0, 0, 0, 0, 0, 0, 0,
+        // 0x01
+        0, 0, 0, 0, 0, 0, 0, 1,
+        // 0x03
+        0, 0, 0, 0, 0, 0, 1, 1,
+        // 0x12
+        0, 0, 0, 1, 0, 0, 1, 0,
+        // 0xFF
+        1, 1, 1, 1, 1, 1, 1, 1
+    };
+    std::vector<bool> actual_bits;
+    libzerocash::convertBytesVectorToVector(bytes, actual_bits);
+    BOOST_CHECK(actual_bits == expected_bits);
+}
+
+BOOST_AUTO_TEST_CASE( testConvertVectorToBytesVector ) {
+    std::vector<unsigned char> expected_bytes = {0x00, 0x01, 0x03, 0x12, 0xFF};
+    std::vector<bool> bits = {
+        // 0x00
+        0, 0, 0, 0, 0, 0, 0, 0,
+        // 0x01
+        0, 0, 0, 0, 0, 0, 0, 1,
+        // 0x03
+        0, 0, 0, 0, 0, 0, 1, 1,
+        // 0x12
+        0, 0, 0, 1, 0, 0, 1, 0,
+        // 0xFF
+        1, 1, 1, 1, 1, 1, 1, 1
+    };
+    std::vector<unsigned char> actual_bytes;
+    libzerocash::convertVectorToBytesVector(bits, actual_bytes);
+    BOOST_CHECK(actual_bytes == expected_bytes);
+}
+
+BOOST_AUTO_TEST_CASE( testConvertIntToBytesVector ) {
+    uint64_t val;
+    std::vector<unsigned char> expected;
+    std::vector<unsigned char> bytes(8);
+
+    val = 0ULL;
```

Comment:
ULL used to be 128 bits on Alpha (may it rest in peace). I believe that it's no longer allowed to be 128 bits in C99, but I have no idea about C++.